src/components/pages/home/transaction-history.tsx

```diff
@@ -28,54 +29,27 @@ type DataPoint = {
   name: string;
   date: string;
   price: number;
+  transactionsCount: number;
 };
 
-const CustomTooltip = ({ active, payload }: any) => {
+const CustomTooltip = ({ active, payload }: TooltipProps<number, string>) => {
   if (active && payload && payload.length) {
+    const { date, price, transactionsCount } = payload[0].payload;
     return (
       <div className="rounded-md border bg-popover px-3 py-1.5 text-sm text-popover-foreground shadow-md">
         <p className="text-xs">
-          {format(parseISO(payload[0].payload.date), "EEEE, MMMM d, yyyy")}
+          {format(parseISO(date), "EEEE, MMMM d, yyyy")}
         </p>
-        <p>Price: {formatPrice(payload[0].payload.price)}</p>
+        <p>Price: {formatPrice(price)}</p>
+        <p>Transactions: {transactionsCount}k</p>
       </div>
     );
   }
   return null;
 };
 
-const TransactionHistory = () => {
-  const [data, setData] = useState<DataPoint[]>([]);
-
-  useEffect(() => {
-    const fetchData = async () => {
-      const dates = Array.from({ length: 15 }, (_, i) => {
-        const date = subDays(new Date(), 14 - i);
-        return {
-          name: format(date, "MMM d"),
-          formattedDate: format(date, "yyyy-MM-dd"),
-        };
-      });
-
-      const responses = await Promise.all(
-        dates.map(async (dateObj) => {
-          const response = await fetch(
-            `https://api.coinbase.com/v2/prices/OP-USD/spot?date=${dateObj.formattedDate}`,
-          );
-          const json = await response.json();
-          return {
-            name: dateObj.name,
-            date: dateObj.formattedDate,
-            price: parseFloat(json.data.amount),
-          };
-        }),
-      );
-
-      setData(responses);
-    };
-
-    fetchData();
-  }, []);
+const TransactionHistory = ({ history }: { history: DataPoint[] }) => {
+  const [data, setData] = useState<DataPoint[]>(history);
```

Comment:
I think this is no longer necessary and you can use the history prop directly.

Reply:
done chief